http-clients/java-nio-client/src/main/java/software/amazon/awssdk/http/nio/java/JavaNioAsyncHttpClient.java

```diff
@@ -0,0 +1,251 @@
+    }
+
+
+    public static Builder builder() {
+        return new DefaultBuilder();
+    }
+
+
+    public void close() {
+    }
+
+    @Override
+    public String clientName() {
+        return CLIENT_NAME;
+    }
+
+
+    /**
+     * Builder that allows configuration of the Java NIO HTTP implementation. Use {@link #builder()} to configure
+     * and construct a Java Http Client.
+     */
+    public interface Builder extends SdkAsyncHttpClient.Builder<JavaNioAsyncHttpClient.Builder> {
+
+        /**
+         * Maximum number of allowed concurrent requests.
+         *
+         * <p>
+         * If the maximum number of concurrent requests is exceeded they may queued in the HTTP Client and can
+         * cause latencies. If the client is overloaded enough such that the pending requests queue fills up,
+         * then subsequent requests may be rejected or time out.
+         * </p>
+         *
+         * @param maxConcurrency New value for max concurrency.
+         * @return This builder for method chaining.
+         */
+        Builder maxConcurrency(Integer maxConcurrency);
+
+
+        /**
+         * The amount of time to wait for a connection before an exeception is thrown.
+         *
+         * @param connectionTimeout timeout duration.
+         * @return This builder for method chaining.
+         */
+        Builder connectionTimeout(Duration connectionTimeout);
+
+
+        /**
+         * Sets the HTTP protocol to use (i.e. HTTP/1.1 or HTTP/2). Not all services support HTTP/2.
+         *
+         * @param protocol Protocol to use.
+         * @return This builder for method chaining.
+         */
+        Builder protocol(Protocol protocol);
+
+        /**
+         * Sets the SSL related parameters (e.g. Protocols, CipherSuites, ApplicationProtocols etc.) via SSLParameters
+         * object.
+         *
+         * @param sslParameter SSLParameters object.
+         * @return This builder for method chaining.
+         */
+        Builder configureSsl(SSLParameters sslParameter);
+
+        /**
+         * Sets the proxy related configurations (e.g. URI, port number etc.).
+         *
+         * @param proxyconfiguration ProxyConfiguration object.
+         * @return This builder for method chaining.
+         */
+        Builder proxyConfig(ProxyConfiguration proxyconfiguration);
+
+        /**
+         * Sets the amount of time to wait for a response before timeout.
+         *
+         * @param responseTimeout timeout duration.
+         * @return This builder for method chaining.
+         */
+        Builder responseTimeout(Duration responseTimeout);
+
+
+        /**
+         * Sets the numberOfThreads that the executor can hold in the threadspool.
+         *
+         * @param numberOfThreads an integer as the number of threads will be used in the executor.
+         * @return This builder for method chaining.
+         */
+        Builder numberOfThreads(Integer numberOfThreads);
```

Comment:
`numOfThreads`

Reply:
There is a similar field in the SdkEventLoopGroup of Netty named exactly the same as numberOfThreads, shouldn't we keep it the same name here? That's the reason why I changed it into numberOfThreads here. Please let me know if I still need to change it into numOfThreads, thanks!!

Reply:
Ohhh, I must remember it wrongly.